src/librustc_mir/build/matches/test.rs

```diff
@@ -140,6 +144,30 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
         }
     }
 
+    pub fn add_variants_to_switch<'pat>(&mut self,
+                                     test_lvalue: &Lvalue<'tcx>,
+                                     candidate: &Candidate<'pat, 'tcx>,
+                                     variants: &mut Vec<bool>)
+                                     -> bool
+    {
+        let match_pair = match candidate.match_pairs.iter().find(|mp| mp.lvalue == *test_lvalue) {
+            Some(match_pair) => match_pair,
+            _ => { return false; }
+        };
+
+        match *match_pair.pattern.kind {
+            PatternKind::Variant { adt_def: _ , variant_index,  .. } => {
+                // Do I need to look at the PatternKind::Variant subpatterns?
```

Comment:
This comment reads like a "FIXME" but I think it's sort of a rhetorical question? Maybe something like "We have a pattern testing for the variant `variant_index`, so set the corresponding index to true"